### DISCUSSION TITLE
Translations not working in Calendar Layout

### DISCUSSION THREAD

**Discussion by azrikahar** [directus/directus#9559] [Feature Requests]
When we set translations fields as the display template, in this case `translations.title`, it shows up as undefined:

![image](https://user-images.githubusercontent.com/42867097/140617661-a0bb7e90-3aba-45b9-b960-ec8ba56ce9eb.png)

### To Reproduce

- Create an Articles collection with translations field

- Add `publish_date` datetime input to `articles` collection (for calendar view later)

- Add `title` string input to `articles_translations` collection

- Create new Article with publish date & title

- Go to Articles, set as calendar layout, set Display template as `Title`, start & end date field as `Publish Date`:
   ![image](https://user-images.githubusercontent.com/42867097/140617594-d8e57ca7-2521-4671-8435-8650a1b4b255.png)

   Sample screenshot of layout options:

   ![image](https://user-images.githubusercontent.com/42867097/140617575-74abfec8-e26d-4737-a5a9-9bce2a71200b.png)

### Errors Shown

No error shown.

### What version of Directus are you using?

v9.0.0 (main branch)

### What version of Node.js are you using?

16.13.0

### What database are you using?

PostgreSQL 13.4

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

Docker

**Comment by azrikahar** [comment]
Preliminary investigation:

Items are fetched with `translations.title` as expected:

![image](https://user-images.githubusercontent.com/42867097/140617767-254e3c2e-228c-4377-80d2-5067846315a9.png)

This is due to getFieldsFromTemplate here:

https://github.com/directus/directus/blob/be2d7b267d5dbc0212f64473774e7f1709a4a9c2/app/src/layouts/calendar/index.ts#L137-L145

However translations is an array, so it's probably doing `translation.title` instead of `translation[N].title` when rendering the title here:

https://github.com/directus/directus/blob/be2d7b267d5dbc0212f64473774e7f1709a4a9c2/app/src/layouts/calendar/index.ts#L326

since it failed to do so, it returns the latter `undefined` here.

**Comment by rijkvanzanten** [comment]
This is not a bug in the app, as it's not supposed to work like that. A one to many is an array of items, so if you try to read a nested field, there is nothing to read, as it's an array of items. The related-values display would show "12 Items" with a dropdown to show the preview template for each individual item; a similar thing needs to be implemented for in here. Ideally, we can use the regular render-template handlers inside the event placeholder. I'll move this to a discussion as it's technically a new feature request 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
